aexpect/session_ops.py

```diff
@@ -0,0 +1,598 @@
+            if pattern.match(filename)]
+
+
+def cat(session, filename):
+    """
+    Get contents of a text file from a session.
+
+    :param session: session to run the command on
+    :type session: ShellSession
+    :param str filename: full path of file
+    :returns: file contents
+    :rtype: str
+    :raises: :py:class:`RuntimeError` if cat command fails
+
+    Should only be used for very short files without tabs or other fancy
+    contents. Otherwise better download file or use some other method.
+    """
+    cmd = 'cat ' + quote(filename)
+    status, output = session.cmd_status_output(cmd)
+    if output:
+        output = output.strip()
+    if status != 0:
+        raise RuntimeError('Failed to cat {}: {}'.format(filename, output))
+    return output
+
+
+def tempdir(session):
+    """
+    Create a temporary directory in a safe way through a session.
+
+    :param session: session to run the command on
+    :type session: ShellSession
+    :returns: name of temporary directory
+    :rtype: str
+    :raises: :py:class:`RuntimeError` if mktemp returns non-zero exit status
+
+    Calls `mktemp -d`, refer to `man` for more info.
+    """
+    status, output = session.cmd_status_output('mktemp -d')
+    if output:
+        output = output.strip()
+    if status != 0:
+        raise RuntimeError('Failed to create temporary directory: {}'
+                           .format(output))
+    return output
+
+
+def tempfile(session):
+    """
+    Create a temporary file in a safe way through a session.
+
+    :param session: session to run the command on
+    :type session: ShellSession
+    :returns: name of temporary file
+    :rtype: str
+    :raises: :py:class:`RuntimeError` if mktemp returns non-zero exit status
+
+    Calls `mktemp`, refer to `man` for more info.
+    """
+    status, output = session.cmd_status_output('mktemp')
+    if output:
+        output = output.strip()
+    if status != 0:
+        raise RuntimeError('Failed to create temporary file: {}'
+                           .format(output))
+    return output
+
+
+def move(session, source, target):
+    """
+    Move file or directory from source to target / Rename source to target.
+
+    :param session: session to run the command on
+    :type session: ShellSession
+    :param str source: full path of existing file or directory
+    :param str target: full path of new file or directory
+    :raises: :py:class:`RuntimeError` if mv returns non-zero exit status
+
+    Calls `mv source target` through a session. See `man mv` for what source
+    and target can be and what behavior to expect.
+    """
+    cmd = 'mv {} {}'.format(quote(source), quote(target))
+    status, output = session.cmd_status_output(cmd)
+    if status != 0:
+        raise RuntimeError('Failed to mv {} to {}: {}'
+                           .format(source, target,
+                                   output.strip() if output else ''))
+
+
+def copy(session, source, target):
```

Comment:
Like grep, we do have some 'flags', what about introduce those flags here too ?

Reply:
I could and for sure there are lots of arguments that could be extended for many functions here. You might have already guessed that I accumulated all these functions from our our various sources and they might slightly differ in style and approach as they were written by different collaborator devs. So I am not sure the extra step for a single function is worth it as my intention was rather that future contributors could gradually keep extending these functions similarly to the way they started with avocado's ssh utility.